src/main/java/org/zalando/nakadi/filters/LoggingFilter.java

```diff
@@ -0,0 +1,45 @@
+package org.zalando.nakadi.filters;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.security.Principal;
+import java.util.Optional;
+
+@Component
+public class LoggingFilter extends OncePerRequestFilter {
+
+    private static final Logger LOG = LoggerFactory.getLogger(LoggingFilter.class);
+
+    @Override
+    protected void doFilterInternal(final HttpServletRequest request,
+                                    final HttpServletResponse response, final FilterChain filterChain)
+            throws ServletException, IOException
+    {
+        final long start = System.currentTimeMillis();
+
+        //execute request
+        try {
+            filterChain.doFilter(request, response);
+        } finally {
```

Comment:
how it will log our async consumption requests? In the beginning of request or when the stream is finished?